app/Http/Controllers/ShopifyController.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Http\Interfaces\DataProviderInterface;
+use GuzzleHttp\Client;
+use Psr\Http\Message\ResponseInterface;
+use \Symfony\Component\HttpFoundation\Response;
+
+class ShopifyController extends Controller
+{
+    public function getProducts(): array
+    {
+        $response = $this->fetchShopifyProducts();
+
+        return json_decode($response->getBody()->getContents(),true);
+    }
+
+   public function showProduct(DataProviderInterface $printData): Response
+   {
+       return $printData->transformData($this->getProducts());
+   }
+
+   private function fetchShopifyProducts(): ResponseInterface
```

Comment:
should be done by the ShouldInteractWithShopify interface and BasicShopifyInteractor service